src/pyfmi/simulation/assimulo_interface.pyx

```diff
@@ -640,10 +638,9 @@ cdef class FMIODE2(cExplicit_Problem):
         self._sparse_representation = False
         self._with_jacobian = with_jacobian
 
-        if self._logging and isinstance(result_handler, ResultHandlerBinaryFile):
-            self._logging_to_mat = 1
-        else:
-            self._logging_to_mat = 0
+        ## If result handler support is available, logging turns into dynamic_diagnostics
+        self._logging_as_dynamic_diagnostics = self._logging and result_handler.supports.get("dynamic_diagnostics")
+        self._number_of_diagnostics_variables = number_of_diagnostics_variables
```

Comment:
the number of diagnostics variables as previously taken from the results_handler.